src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,207 @@
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
+ * `enveloping cube` and `enveloping sphere` refer to the bounding surfaces of
+ * the outer Sphere of ten wedges making up the last layer of the domain.
+ *
+ * The plane along which the domain divides into two hemispheres is referred
+ * to as the "cutting plane".
+ */
+
+template <typename TargetFrame>
+class BinaryCompactObject : public DomainCreator<3, TargetFrame> {
+ public:
+  struct InnerRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct XCoordObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorA {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct InnerRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct XCoordObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorB {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct RadiusOuterCube {
+    using type = double;
+    static constexpr OptionString help = {
+        "Radius circumscribing the layer of Frustums."};
+  };
+
+  struct RadiusOuterSphere {
+    using type = double;
+    static constexpr OptionString help = {"Radius of the entire domain."};
+  };
+
+  struct InitialRefinement {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial refinement level. Applied to each dimension."};
+  };
+
+  struct InitialGridPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial number of grid points in each dim per element."};
+  };
+
+  struct UseEquiangularMap {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Use equiangular instead of equidistant coordinates."};
+    static type default_value() { return true; }
+  };
+
+  struct OriginPreimage {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of center of mass before translation."};
```

Comment:
The center of mass is computed by the user, the `before translation` refers to the fact that the location of the center of mass is provided in the initial coordinates before the translation map is applied to the inner layers. In the new coordinates, the location of the center of mass is the origin.

In this domain the [cutting plane no longer stays planar](https://github.com/sxs-collaboration/spectre/blob/a7396ac2ae1992c09997cc1fa797c8eeaf2864d0/docs/Images/Domain/DomainCreators/binary_compact_object_domain.png) once the translation maps are applied to the inner layers. In the initial coordinates, the cutting plane hits the x-axis at x=0. In the new coordinates after applying the COM translation, the cutting plane hits the x-axis at x'=-COM (so that x=COM is sent to x'= 0).

For the cutting plane to hit the x-axis at `0.5*(XCoordObjectA+XCoordObjectB)`, the `OriginPreimage` option should then be set to `-0.5*(XCoordObjectA+XCoordObjectB)`.